api/src/main/java/io/smallrye/stork/api/LoadBalancer.java

```diff
@@ -22,4 +22,10 @@
      *         are deemed invalid for some reason
      */
     ServiceInstance selectServiceInstance(Collection<ServiceInstance> serviceInstances);
+
+    /**
+     *
+     * @return true if the selectServiceInstance method can safely be called in parallel
+     */
+    boolean isThreadSafe();
```

Comment:
Not sure I like the name. Your implementation can be thread-safe, but still does not work as expected on concurrent calls.

Reply:
We should have a default implementation.

Reply:
Maybe sth like this?
```java
default boolean requiresStrictStatCollection() {
   return true;
}
```

Reply:
Yes, looks better. 

`true` by default is the safest, so +1.